ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+
+function imap_rfc822_parse_adrlist(string $address_string, string $default_host): array {}
+
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_body($stream_id, int $msg_no, int $options = 0) {}
+
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_bodystruct($stream_id, int $msg_no, string $section) {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchbody($stream_id, int $msg_no, string $section, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchmime($stream_id, int $msg_no, string $section, int $options = 0) {}
+
+
+/**
+ * @param resource $stream_id
+ * @todo: should $file be `resouce|string`? it looks like it tries to accept anything?
+ */
+function imap_savebody($stream_id, $file, int $msg_no, string $section = '', int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return string|false
+ */
+function imap_fetchheader($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_fetchstructure($stream_id, int $msg_no, int $options = 0) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_gc($stream_id, int $flags): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_expunge($stream_id): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_delete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_undelete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_check($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_listscan($stream_id, string $ref, string $pattern, string $content) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_copy($stream_id, string $msglist, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_move($stream_id, string $sequence, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_mail_compose(array $envelope, array $body) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_createmailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_renamemailbox($stream_id, string $old_name, string $new_name): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_deletemailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_subscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_unsubscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_append($stream_id, string $folder, string $message, string $options = '', string $internal_date = ''): bool {}
```

Comment:
It seems `$options` and `$internal_date` do not really have a default value, so:
```suggestion
function imap_append($stream_id, string $folder, string $message, string $options = UNKNOWN, string $internal_date = UNKNOWN): bool {}
```